### PR TITLE
Integrate EmailChecker in the app

### PR DESCRIPTION
**Pull request by koke**
As mentioned on [a previous comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/5267#discussion_r61866381), here's a newer Swift version of EmailChecker.

[EmailChecker](https://github.com/wordpress-mobile/EmailChecker) was born as an experiment to share code between iOS and Android. Recently the android version was rewritten in Kotlin and moved to [EmailChecker-Android](https://github.com/wordpress-mobile/EmailChecker-Android). Around the same time, I ported that new version to Swift in [EmailChecker-ios](https://github.com/koke/EmailChecker-ios).

Since this is a really small library that doesn't seem to be used outside the WordPress app, I think it's best to integrate it in the app directly. So I've removed the pod dependency and included the new Swift version inside the app.

I've also renamed it a bit so it matches the intention better: `EmailChecker` sounds like we might be validating emails, when it's actually just about checking for typos.

`EmailChecker.suggestDomainCorrection(_:)` is now `EmailTypoChecker.guessCorrection(email:)`.

`+ [EmailChecker suggestDomainCorrection:]` is now `+ [EmailTypoChecker guessCorrectionForEmail:]`

As this changes the `Podfile`, I anticipate conflicts with #5357 and the ongoing Cocoapods 1.0 work.

Needs review: @aerych

### CODE REVIEW

File: WordPress/Classes/Extensions/String+Helpers.swift [hunk 1]
```diff
@@ -30,4 +30,13 @@ extension String {
         }
         return self
     }
+
+    /// Returns a string without the character at the specified index.
+    /// This is a non-mutating version of `String.removeAtIndex(_:)`.
+    func removingCharacterAtIndex(index: Index) -> String {
```
Comment (aerych) [inline WordPress/Classes/Extensions/String+Helpers.swift]: Maybe `removeCharacterAtIndex` vs `removingChar...`  The gerund tense just sounds weird to me.

Comment (koke) [inline WordPress/Classes/Extensions/String+Helpers.swift]: I'm living in the future 😉 https://swift.org/documentation/api-design-guidelines/#name-according-to-side-effects
That said, in Swift 3 the mutating method will be `String.remove(at:)`, so this should be `String.removing(at:)`

**Comment by aerych** [comment]
Hiya Koke :)  Thanks for wrangling this one.  The branch builds, the test suite passes, and the signup form (new and old) is correcting typos as expected.  One minor comment that might just be a matter of taste. Otherwise :shipit: when ready.

**Comment by koke** [comment]
Thanks, I corrected the name to match Swift 3 style and added a bit of documentation. Merging now

**Pull request merged**